core/appeal/service.go

```diff
@@ -525,6 +531,14 @@ func (s *Service) UpdateApproval(ctx context.Context, approvalAction domain.Appr
 		}
 
 		if err := s.Update(ctx, appeal); err != nil {
+
```

Comment:
try handling the error by comparing with `gorm`'s errors or using postgres errors, see this for reference https://gorm.io/docs/error_handling.html

Reply:
also the error handling should happen in the repository, here in the service we should only check using `errors.Is`